### PR TITLE
Add Push Notifications command

### PR DESCRIPTION
**Pull request by codeofdiego**
This PR adds the ability to set up Push Notifications as an independent command to be used after the user executes the initial create setup. It performs the following changes in the project:
- Add React Native Firebase and messaging dependencies
- Copies a template directory with a new `useNotifications` hook and `requestNotificationPermission` utility
- Updates the `App.tsx` file to import and run the hook
- Adds the native folders to the `.gitignore` file
- Adds required config object to the `app.json` file.

To achieve some of these changes 3 new utility functions were added:
- `readAppJson` - returns the `app.json` file as a parsed json object
- `addAppJsonConfig` - deep merges a provided [ExpoConfig](https://docs.expo.dev/versions/latest/config/app/) config with the existing content of the `app.json` file
- `injectHooks` - takes a list of hooks and their imports, and injects them in the `App.tsx` file

Additionally, this PR includes new rules for import organizations in the `.eslintrc.js.eta` template, this was necessary to give us the ability to reorganize imports when we inject code/imports in existing files. This will be a repeating scenario as we add more commands to Betl.

**Preview**
| Collapsed | Expanded |
|:---:|:---:|
| Screenshot 2024-02-27 at 14 27 21 | Screenshot 2024-02-27 at 14 27 24 |

### CODE REVIEW

File: src/commands/__tests__/notifications.test.ts [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+import addDependency from '../../util/addDependency';
+import copyTemplateDirectory from '../../util/copyTemplateDirectory';
+import { input } from '@inquirer/prompts';
+
+vi.mock('@inquirer/prompts', () => ({
+  input: vi.fn(),
+}));
+vi.mock('../../util/addDependency');
+vi.mock('../../util/copyTemplateDirectory');
+vi.mock('../../util/getProjectDir', () => ({ default: vi.fn(() => './') }));
```
[the remark this answers is not in the document]
Comment (codeofdiego) [reply src/commands/__tests__/notifications.test.ts]: Correct, I guess that at some point I experimented with it and left the mock there. Just removed it.

File: src/commands/notifications.ts [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+import copyTemplateDirectory from '../util/copyTemplateDirectory';
+import exec from '../util/exec';
+import isExpo from '../util/isExpo';
+import addExpoConfig from '../util/addExpoConfig';
+import injectHooks from '../util/injectHooks';
+import readAppJson from '../util/readAppJson';
+import { input } from '@inquirer/prompts';
+import commit from '../util/commit';
+
+const handleCommitError = (error: { stdout: string }) => {
```
[the remark this answers is not in the document]
Comment (codeofdiego) [reply src/commands/notifications.ts:18]: The goal of this handler was specific to the case where the `notifications` command is run two times. There would be `nothing to commit` and the command would fail midway without trying the next steps that may or may not cause changes (depending on any user-made changes).

This happened more often when I was testing the command repeatedly during development but I figured it would be good to handle the edge case.

I removed the ESLint override, which was unnecessary.

The intention there is for it to just handle the `nothing to commit` case which has a `stdout`, every other type of error should throw as usual.

**Review by stevehanson — APPROVED** [review]
Great work!
